### ISSUE TITLE
my CNN models print NAN. what is reason??? i don't know what is it.

### ISSUE THREAD

**Issue by pervin0527** [tensorflow/tensorflow#23349]
I made a CNN model.
All results are output as "nan".
What's wrong? Please help me.

here is my repo
https://github.com/pervin0527/pervin0527 

def get_filename_set(data_set):
    labels = []
    filename_set = []

    with open(FLAGS.data_dir + '/labels.txt') as f:
        for line in f:
            inner_list = [elt.strip() for elt in line.split(',')]
            labels += inner_list

    for i, lable in enumerate(labels):
        list = os.listdir(FLAGS.data_dir  + '/' + data_set + '/' + lable)
        for filename in list:
            filename_set.append([i, FLAGS.data_dir  + '/' + data_set + '/' + lable + '/' + filename])

    random.shuffle(filename_set)
    return filename_set

def read_jpeg(filename):
    value = tf.read_file(filename)
    decoded_image = tf.image.decode_jpeg(value, channels=FLAGS.depth)
    resized_image = tf.image.resize_images(decoded_image, FLAGS.raw_height, FLAGS.raw_width)
    resized_image = tf.cast(resized_image, tf.uint8)

    return resized_image

def convert_images(sess, data_set):
    filename_set = get_filename_set(data_set)

    with open('./data/' + data_set + '_data.bin', 'wb') as f:
        for i in range(0, len(filename_set)):
            resized_image = read_jpeg(filename_set[i][1])

            try:
                image = sess.run(resized_image)
            except Exception as e:
                print e.message
                continue

            #plt.imshow(np.reshape(image.data, [FLAGS.raw_height, FLAGS.raw_width, FLAGS.depth]))
            #plt.show()

            print i, filename_set[i][0], image.shape
            f.write(chr(filename_set[i][0]))
            f.write(image.data)

def read_raw_images(sess, data_set):
    filename = ['./data/' + data_set + '_data.bin']
    filename_queue = tf.train.string_input_producer(filename)

    record_bytes = (FLAGS.raw_height) * (FLAGS.raw_width) * FLAGS.depth + 1
    reader = tf.FixedLengthRecordReader(record_bytes=record_bytes)
    key, value = reader.read(filename_queue)
    record_bytes = tf.decode_raw(value, tf.uint8)

    tf.train.start_queue_runners(sess=sess)

    for i in range(0, 100):
        result = sess.run(record_bytes)
        print i, result[0]
        image = result[1:len(result)]

        #plt.imshow(np.reshape(image, [FLAGS.raw_height, FLAGS.raw_width, FLAGS.depth]))
        #plt.show()

FLAGS = tf.app.flags.FLAGS

def weight_variable(shape, name):
    with tf.device('/cpu:0'):
        initial = tf.truncated_normal(shape, stddev=0.1)
        var = tf.Variable(initial, name=name)
    return var

def bias_variable(shape, name):
    with tf.device('/cpu:0'):
        initial = tf.constant(0.1, shape=shape)
        var = tf.Variable(initial, name=name)
    return var

def conv2d(x, W):
    return tf.nn.conv2d(x, W, strides=[1, 1, 1, 1], padding='SAME')

def max_pool(x, height, width):
    return tf.nn.max_pool(x, ksize=[1, height, width, 1], strides=[1, height, width, 1], padding='SAME')

def make_network(images, labels, keep_prob):
    num_conv = FLAGS.num_conv
    kernel_size = FLAGS.kernel_size
    pool_size = FLAGS.pool_size
    num_map = FLAGS.num_map
    num_fc_layer = FLAGS.num_fc_layer
    num_fc_input = FLAGS.num_fc_input

    height = FLAGS.height
    width = FLAGS.width
    prev_num_map = FLAGS.depth
    h_pool = images

    for i in range(num_conv):
        W_conv = weight_variable([kernel_size, kernel_size, prev_num_map, num_map], 'W_conv' + str(i+1))
        b_conv = bias_variable([num_map], 'b_conv' + str(i+1))
        h_conv = tf.nn.relu(conv2d(h_pool, W_conv) + b_conv)
        h_pool = max_pool(h_conv, pool_size, pool_size)
        prev_num_map = num_map
        num_map *= 2
        height /= 2
        width /= 2

    num_map /= 2
    h_fc_input = tf.reshape(h_pool, [-1, height * width * num_map])
    prev_num_fc_input = height * width * num_map

    for i in range(num_fc_layer):
        W_fc = weight_variable([prev_num_fc_input, num_fc_input], 'W_fc' + str(i+1))
        b_fc = bias_variable([num_fc_input], 'b_fc' + str(i+1))
        h_fc = tf.nn.relu(tf.matmul(h_fc_input, W_fc) + b_fc)
        h_fc_input = tf.nn.dropout(h_fc, keep_prob)
        prev_num_fc_input = num_fc_input
        num_fc_input /= 2

    num_fc_input *= 2
    W_fc = weight_variable([num_fc_input, FLAGS.num_class], 'W_fc' + str(i+2))
    b_fc = bias_variable([FLAGS.num_class], 'b_fc' + str(i+2))

    hypothesis = tf.matmul(h_fc_input, W_fc) + b_fc
    cross_entropy = tf.reduce_mean(tf.nn.sparse_softmax_cross_entropy_with_logits(hypothesis, labels))
    train_step = tf.train.AdamOptimizer(FLAGS.learning_rate).minimize(cross_entropy)

    return hypothesis, cross_entropy, train_step

**Comment by ymodak** [comment]
This question is better asked on [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow) since it is not a bug or feature request. There is also a larger community that reads questions there.

If you think we've misinterpreted a bug, please comment again with a clear explanation, as well as all of the information requested in the [issue template](https://github.com/tensorflow/tensorflow/issues/new). Thanks!
